staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -83,29 +83,82 @@ func (g *Cloud) ToInstanceReferences(zone string, instanceNames []string) (refs
 }
 
 // NodeAddresses is an implementation of Instances.NodeAddresses.
-func (g *Cloud) NodeAddresses(_ context.Context, _ types.NodeName) ([]v1.NodeAddress, error) {
-	internalIP, err := metadata.Get("instance/network-interfaces/0/ip")
-	if err != nil {
-		return nil, fmt.Errorf("couldn't get internal IP: %v", err)
+func (g *Cloud) NodeAddresses(ctx context.Context, nodeName types.NodeName) ([]v1.NodeAddress, error) {
+	timeoutCtx, cancel := context.WithTimeout(ctx, 1*time.Hour)
+	defer cancel()
+
+	instanceName := string(nodeName)
+
+	if g.useMetadataServer {
+
+		if g.isCurrentInstance(instanceName) { // Try to use metadata server first
+
+			nics, err := metadata.Get("instance/network-interfaces/")
+			if err != nil {
+				return nil, fmt.Errorf("couldn't get network interfaces: %v", err)
+			}
+
+			nicsArr := strings.Split(strings.TrimSpace(nics), "/")
```

Comment:
If I understand correctly, the metadata returns directory as one item per line...I understand there is a `/` suffix, but doing this way is kind of confusing.
```suggestion
			nicsArr := strings.Split(nics, "/\n")
```